kore/src/Kore/Step/EquationalStep.hs

```diff
@@ -456,3 +460,21 @@ matchRule sideCondition initial rule = do
             "Could not match patterns"
             pattern1
             pattern2
+
+evaluateRequires
+    :: forall unifier variable
+    .  SimplifierVariable variable
+    => MonadUnify unifier
+    => SideCondition variable
+    -> Condition variable
+    -> unifier (Condition variable)
+evaluateRequires sideCondition requires = do
+    _requires' <-
+        simplifyPredicate sideCondition Nothing requires
+        & withoutAxioms
+        & Unifier.maybeUnifierT
+        & maybeT (return requires) Unifier.scatter
+    deepseq _requires' $ simplifyPredicate sideCondition Nothing
```

Comment:
Why this `deepseq`?

Reply:
Ah, sorry, I requested review but forgot to push my latest version! That is left over from testing. :sweat_smile: